curriculum/challenges/english/08-coding-interview-prep/data-structures/breadth-first-search.english.md

```diff
@@ -9,13 +9,20 @@ forumTopicId: 301622
 <section id='description'>
 So far, we've learned different ways of creating representations of graphs. What now? One natural question to have is what are the distances between any two nodes in the graph? Enter <dfn>graph traversal algorithms</dfn>.
 <dfn>Traversal algorithms</dfn> are algorithms to traverse or visit nodes in a graph. One type of traversal algorithm is the breadth-first search algorithm.
-This algorithm starts at one node, first visits all its neighbors that are one edge away, then goes on to visiting each of their neighbors and so on until all nodes have been reached.
+This algorithm starts at one node and visits all its neighbors that are one edge away. It then goes on to visit each of their neighbors and so on until all nodes have been reached.
+
+An important data structure that will help implement the breadth-first search algorithm is the queue. This is an array where you can add elements to one end and remove elements from the other end. This is also known as a <dfn>FIFO</dfn> or <dfn>First-In-First-Out</dfn> data structure.
+
 Visually, this is what the algorithm is doing.
 <img class='img-responsive' src='https://camo.githubusercontent.com/2f57e6239884a1a03402912f13c49555dec76d06/68747470733a2f2f75706c6f61642e77696b696d656469612e6f72672f77696b6970656469612f636f6d6d6f6e732f342f34362f416e696d617465645f4246532e676966'>
+
+The grey shades represent a node getting added into the queue and the black shades represent a node getting removed from the queue. See how everytime a node gets removed from the queue all their neighbors get added into the queue.
+
 To implement this algorithm, you'll need to input a graph structure and a node you want to start at.
-First, you'll want to be aware of the distances from the start node. This you'll want to start all your distances initially some large number, like <code>Infinity</code>. This gives a reference for the case where a node may not be reachable from your start node.
+
+First, you'll want to be aware of the distances from the start node. You'll want to start all your distances initially with some large number, like <code>Infinity</code>. This gives a reference for the case where a node may not be reachable from your start node.
```

Comment:
Distance not defined yet and "initially" is redundant when starting the sentence with "You'll want to start..."

```suggestion
First, you'll want to be aware of the distances from, or number of edges away from, the start node. You'll want to start all your distances with some large number, like <code>Infinity</code>. This prevents counting issues for when a node may not be reachable from your start node.
```